tests/TestCase.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * ImpressCMS Base Test Case
+ *
+ * Base test case class for all ImpressCMS tests.
+ * Provides common functionality and setup for testing ImpressCMS components.
+ *
+ * @copyright	https://www.impresscms.org/ The ImpressCMS Project
+ * @license		MIT
+ * @category	ICMS
+ * @package		Tests
+ * @since		2.1
+ */
+
+namespace Tests;
+
+use PHPUnit\Framework\TestCase as BaseTestCase;
+
+abstract class TestCase extends BaseTestCase
+{
+    /**
+     * Setup the test environment
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // Additional setup can be added here
+        // For example, resetting static state, clearing caches, etc.
+    }
+
+    /**
+     * Clean up the test environment
+     *
+     * @return void
+     */
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        // Clean up any test artifacts
+    }
+
+    /**
+     * Assert that a string contains valid SQL
+     *
+     * @param string $sql
+     * @param string $message
+     * @return void
+     */
+    protected function assertValidSql(string $sql, string $message = ''): void
+    {
+        $this->assertNotEmpty($sql, $message ?: 'SQL should not be empty');
+        $this->assertIsString($sql, $message ?: 'SQL should be a string');
+    }
+
+    /**
+     * Assert that a string is a valid LDAP filter
+     *
+     * @param string $filter
+     * @param string $message
+     * @return void
+     */
+    protected function assertValidLdapFilter(string $filter, string $message = ''): void
```

Comment:
to have this in base class... i'm really not sure if is good idea... because LDAP filter shouldn't be tested with startWith and endsWith asserts... I'm not sure how correct way to test it but I'll try with directorytree/ldaprecord parser to test strings... because at current state we really don't know even if that still works...